repo-projects/Directory.Build.targets

```diff
@@ -55,6 +55,11 @@
     <MinimalConsoleLogOutput Condition="'$(MinimalConsoleLogOutput)' == ''">$(BuildInParallel)</MinimalConsoleLogOutput>
   </PropertyGroup>
 
+  <!-- Use the P2P protocol for RepositoryReference items. -->
+  <ItemGroup>
+    <ProjectReference Include="@(RepositoryReference->'%(Identity).proj')" IsRepositoryReference="true" BuildReference="false" />
+  </ItemGroup>
```

Comment:
Now that we have this, I wonder whether we can take advantage of MSBuild's traversal of the project references to replace or simplify the logic here:

https://github.com/dotnet/dotnet/blob/8efffbde026eccfca4eed0b453380dd91993c824/repo-projects/Directory.Build.targets#L58-L79

Does MSBuild provide a representation of the flattened list of dependencies?